src/tests/routes/crates/list.rs

```diff
@@ -262,60 +306,172 @@ fn index_sorting() {
+        // ordering (exact match desc, name asc)
+        let mut sorted = decoded_seeks.to_vec();
+        sorted.sort_by_key(|k| (Reverse(k.0), k.1.to_owned()));
+        assert_eq!(sorted, decoded_seeks);
+        for json in search_both(&anon, query) {
+            assert_eq!(json.meta.total, resp[0].meta.total);
+            for (c, r) in json.crates.iter().zip(&resp) {
+                assert_eq!(c.name, r.crates[0].name);
+            }
+        }
+    }
+
+    // Sort by relevance
+    for query in ["q=foo_sort", "q=sort"] {
+        let (resp, calls) = page_with_seek(&anon, query);
+        assert_eq!(calls, resp[0].meta.total + 1);
+        let decoded_seeks = resp
+            .iter()
+            .filter_map(|cl| {
+                cl.meta
+                    .next_page
+                    .as_ref()
+                    .map(|next_page| (next_page, cl.crates[0].name.to_owned()))
+            })
+            .filter_map(|(q, name)| {
+                let query = url::form_urlencoded::parse(q.trim_start_matches('?').as_bytes())
+                    .into_owned()
+                    .collect::<indexmap::IndexMap<String, String>>();
+                query.get("seek").map(|s| {
+                    let d = decode_seek::<(bool, f32, i32)>(s).unwrap();
+                    (d.0, (d.1 * 1e12) as i64, name)
+                })
+            })
+            .collect::<Vec<_>>();
+        // ordering (exact match desc, rank desc, name asc)
+        let mut sorted = decoded_seeks.clone();
+        sorted.sort_by_key(|k| (Reverse(k.0), Reverse(k.1), k.2.to_owned()));
+        assert_eq!(sorted, decoded_seeks);
+        for json in search_both(&anon, query) {
+            assert_eq!(json.meta.total, resp[0].meta.total);
+            for (c, r) in json.crates.iter().zip(&resp) {
+                assert_eq!(c.name, r.crates[0].name);
+            }
+        }
+    }
 
     // Test for bug with showing null results first when sorting
     // by descending downloads
-    let json = anon.search("sort=recent-downloads");
-    assert_eq!(json.meta.total, 4);
-    assert_eq!(json.crates[0].name, "foo_sort");
-    assert_eq!(json.crates[1].name, "baz_sort");
-    assert_eq!(json.crates[2].name, "bar_sort");
-    assert_eq!(json.crates[3].name, "other_sort");
+    for json in search_both(&anon, "sort=recent-downloads") {
+        assert_eq!(json.meta.total, 4);
+        assert_eq!(json.crates[0].name, "foo_sort");
+        assert_eq!(json.crates[1].name, "baz_sort");
+        assert_eq!(json.crates[2].name, "bar_sort");
+        assert_eq!(json.crates[3].name, "other_sort");
+    }
+    let (resp, calls) = page_with_seek(&anon, "sort=recent-downloads");
+    assert_eq!(resp[0].crates[0].name, "foo_sort");
+    assert_eq!(resp[1].crates[0].name, "baz_sort");
+    assert_eq!(resp[2].crates[0].name, "bar_sort");
+    assert_eq!(resp[3].crates[0].name, "other_sort");
+    assert_eq!(resp[3].meta.total, 4);
+    assert_eq!(calls, 5);
```

Comment:
This code snippet can be safely removed as it's already covered by above test case.